src/base/log.cc

```diff
@@ -148,13 +149,68 @@ void set_logging_dir() {
 std::unique_ptr<Logger> logger(new Logger());
 #endif
 
-void log(const char* const fmt, ...) {
+static const char* to_string(const LogType& type) {
+	switch (type) {
+	case LogType::kInfo:
+		return "INFO";
+	case LogType::kDebug:
+		return "DEBUG";
+	case LogType::kWarning:
+		return "WARNING";
+	case LogType::kError:
+		return "ERROR";
+	default:
+		NEVER_HERE();
+	}
+}
+
+std::vector<std::string> split(const std::string& s) {
+	std::vector<std::string> result;
+	for (std::string::size_type pos = 0, endpos;
+	     (pos = s.find_first_not_of('\n', pos)) != std::string::npos; pos = endpos) {
+		endpos = s.find('\n', pos);
+		result.push_back(s.substr(pos, endpos - pos));
+	}
+	return result;
+}
+
+void log_to_stdout(const LogType type, uint32_t gametime, const char* const fmt, ...) {
 	assert(logger != nullptr);
 
+	// message type
+	char buffer_prefix[8];
+	snprintf(buffer_prefix, sizeof(buffer_prefix), "%s", to_string(type));
+	char buffer_timestamp[32];
+	if (gametime != kNoTimestamp && gametime > 0) {  // timestamp
+		const uint32_t hours = gametime / (1000 * 60 * 60);
+		gametime -= hours * 1000 * 60 * 60;
+		const uint32_t minutes = gametime / (1000 * 60);
+		gametime -= minutes * 1000 * 60;
+		const uint32_t seconds = gametime / 1000;
+		gametime -= seconds * 1000;
```

Comment:
`gametime` might become zero here; if that happens the condition on line 208 will fail

Reply:
Well spotted, fixed the condition